terminal/src/Format.hs

```diff
@@ -133,6 +135,52 @@ format flags (Env inputs) =
     Project projectType paths ->
       formatFilesOnDisk flags projectType paths
 
+validate :: Env -> Task.Task Exit.Format ()
+validate (Env inputs) = do
+  case inputs of
+    Stdin ->
+      do
+        original <- Task.io BS.getContents
+        case formatByteString Parse.Application original of
+          Nothing ->
+            error "TODO: report error"
+          Just formatted ->
+            when (BSL.fromStrict original /= B.toLazyByteString formatted) $
+              Task.throw Exit.FormatValidateNotCorrectlyFormatted
+    Files paths ->
+      validateFiles paths
+    Project _ paths ->
+      validateFiles paths
+
+validateFiles :: [FilePath] -> Task.Task Exit.Format ()
+validateFiles paths = do
+  validationResults <- mapM validateFile paths
+  when (any (== False) validationResults) $
+    Task.throw Exit.FormatValidateNotCorrectlyFormatted
+
+validateFile :: FilePath -> Task.Task Exit.Format Bool
+validateFile path =
+  do
+    exists <- Task.io (Dir.doesFileExist path)
+    if exists
+      then Task.io (validateExistingFile path)
+      else Task.throw (Exit.FormatPathUnknown path)
```

Comment:
This logic can be shared with `formatFilesOnDisk`

Reply:
Might as well just share that logic, then?